esp8266/modesp.c

```diff
@@ -37,535 +37,57 @@
 #include "queue.h"
 #include "user_interface.h"
 #include "espconn.h"
+#include "ip_addr.h"
+#include "spi_flash.h"
 #include "utils.h"
 
-STATIC const mp_obj_type_t esp_socket_type;
-
-typedef struct _esp_socket_obj_t {
-    mp_obj_base_t base;
-    struct espconn *espconn;
-
-    mp_obj_t cb_connect;
-    mp_obj_t cb_recv;
-    mp_obj_t cb_sent;
-    mp_obj_t cb_disconnect;
-
-    uint8_t *recvbuf;
-    mp_uint_t recvbuf_len;
-
-    bool fromserver;
-
-    mp_obj_list_t *connlist;
-} esp_socket_obj_t;
-
-// Due to the onconnect callback not being able to recognize the parent esp_socket,
-// we can have only one esp_socket listening at a time
-// This should be solvable by some PIC hacking
-STATIC esp_socket_obj_t *esp_socket_listening;
-
-STATIC mp_obj_t esp_socket_make_new_base() {
-    esp_socket_obj_t *s = m_new_obj_with_finaliser(esp_socket_obj_t);
-    s->recvbuf = NULL;
-    s->base.type = (mp_obj_t)&esp_socket_type;
-    s->cb_connect = mp_const_none;
-    s->cb_recv = mp_const_none;
-    s->cb_disconnect = mp_const_none;
-    s->cb_sent = mp_const_none;
-    s->fromserver = false;
-    s->connlist = NULL;
-    return s;
-}
-
-// constructor esp_socket(family=AF_INET, type=SOCK_STREAM, proto=IPPROTO_TCP, fileno=None)
-// Arguments ignored as we do not support UDP (yet)
-STATIC mp_obj_t esp_socket_make_new(mp_obj_t type_in, mp_uint_t n_args,
-    mp_uint_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 0, 4, false);
-
-    esp_socket_obj_t *s = esp_socket_make_new_base();
-    s->espconn = m_new_obj(struct espconn);
-    espconn_create(s->espconn);
-
-    s->espconn->reverse = s;
-    // TODO: UDP Support
-    s->espconn->type = ESPCONN_TCP;
-    s->espconn->state = ESPCONN_NONE;
-
-    s->espconn->proto.tcp = m_new_obj(esp_tcp);
-
-    return s;
-}
-
-// method socket.close()
-STATIC mp_obj_t esp_socket_close(mp_obj_t self_in) {
-    esp_socket_obj_t *s = self_in;
-
-    if (esp_socket_listening == s) {
-        esp_socket_listening = NULL;
-    }
-
-    if (s->espconn->state != ESPCONN_NONE && s->espconn->state != ESPCONN_CLOSE) {
-        espconn_disconnect(s->espconn);
-    }
-
-    if (s->connlist != NULL) {
-        mp_obj_list_set_len(s->connlist, 0);
-    }
-
-    return mp_const_none;
-}
-STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp_socket_close_obj, esp_socket_close);
-
-// method socket.__del__()
-STATIC mp_obj_t esp_socket___del__(mp_obj_t self_in) {
-    esp_socket_obj_t *s = self_in;
-
-    esp_socket_close(self_in);
-
-    if (s->fromserver) {
-        espconn_delete(s->espconn);
-    }
-
+STATIC mp_obj_t esp_deepsleep(mp_uint_t n_args, const mp_obj_t *args) {
```

Comment:
We don't yet have a standard for sleep modes (pyboard has pyb.stop(), cc3200 has pyb.Sleep() class, and there is a discussion to put it in the machine module), so this can stay here for now.